libraries/stdlib/src/kotlin/collections/Collections.kt

```diff
@@ -327,12 +327,12 @@ public inline fun <T, K : Comparable<K>> List<T>.binarySearchBy(
 
 /**
  * Searches this list or its range for an element for which [comparison] function returns zero using the binary search algorithm.
- * The list is expected to be sorted into ascending order according to the provided [comparison],
- * otherwise the result is undefined.
+ * The list is expected to be sorted so that the integer return values of the provided [comparison]
+ * form an ascending sequence on the list elements. Otherwise the result is undefined.
  *
  * If the list contains multiple elements for which [comparison] returns zero, there is no guarantee which one will be found.
  *
- * @param comparison function that compares an element of the list with the element being searched.
+ * @param comparison function that maps elements to integer values.
```

Comment:
This is a restatement of the parameter function signature that doesn't give an idea how this function should be implemented.